src/glue/jobs/s3_to_json.py

```diff
@@ -0,0 +1,209 @@
+"""
+S3 to JSON job
+
+Bucket/sort JSON from a zip archive provided by Care Evolution into
+JSON datasets in S3. A JSON dataset is specific to a key prefix
+(for example: json/dataset=EnrolledParticipants/) and only contains
+files which share a similar schema.
+"""
+import datetime
+import io
+import os
+import json
+import logging
+import sys
+import zipfile
+import boto3
+from awsglue.utils import getResolvedOptions
+
+def write_file_to_json_dataset(z, json_path, dataset_identifier,
```

Comment:
Nit: If you use type hints, vscode has the ability to actually highlight where it may think you're using the wrong types.

```
def write_file_to_json_dataset(z: zipfile.Zipfile, json_path: str, dataset_identifier: str...
```